internal/app/machined/main.go

```diff
@@ -138,38 +140,39 @@ func main() {
 		handle(err)
 	}
 
-	// Start the API server.
-	// NB: This MUST be started AFTER the initialize sequence since the system
-	// directories are setup then.
-	go func() {
-		s := &v1alpha1server.Server{
-			Controller: c,
-		}
-
-		l, e := systemlog.New("machined", constants.DefaultLogPath)
-		if e != nil {
-			handle(e)
-		}
-
-		e = factory.ListenAndServe(
-			s,
-			factory.Network("unix"),
-			factory.SocketPath(constants.MachineSocketPath),
-			factory.WithLog("machined ", l),
-		)
-
-		handle(e)
-	}()
-
 	// Perform an installation if required.
 	if err = c.Run(runtime.SequenceInstall, nil); err != nil {
 		handle(err)
 	}
 
+	// Start the machine API.
+	system.Services(c.Runtime()).LoadAndStart(&services.Machined{Controller: c})
+
 	// Boot the machine.
 	if err = c.Run(runtime.SequenceBoot, nil); err != nil {
 		handle(err)
 	}
 
+	// Watch and handle runtime events.
+	c.Runtime().Events().Watch(func(events <-chan machine.Event) {
+		for {
+			select {
+			case event := <-events:
+				if event.Data.TypeUrl == "talos/runtime/"+proto.MessageName(&machine.SequenceEvent{}) {
```

Comment:
I would like to eventually come up with some helpers that could do this kind of checking but this should get us by for now.

Reply:
my preference will be towards passing some interface as event and allow to typecast directly to Go type

protobuf internally requires too much handling imho

Reply:
i agree ... the hard part then becomes having to maintain internal structs and protuf defined ones... i also wonder how structs would work with plugings. Protobuf felt like the only common denominator.